### DISCUSSION TITLE
Let us use {fmt} as inspiration for refactoring base/strings.h

### DISCUSSION THREAD

**Discussion by ignacionr** [microsoft/vcpkg#17802] [Ideas]
****

**Is your feature request related to a problem? Please describe.**
https://github.com/microsoft/vcpkg/blob/master/toolsrc/include/vcpkg/base/strings.h 
is somewhat chaotic; it could use 1) reorganization, 2) a check on correctness, 3) a check on performance and strategies.

**Proposed solution**
Use https://fmt.dev/latest/index.html as inspiration.

**Describe alternatives you've considered**
proposed using {fmt} as it comes, but it makes sense to not carry dependencies as a package manager

**Additional context**
the file seems to have been constructed just as needs arouse (which is fine); it maybe time for it to get some styling and a health checkup, especially as the expansion in package number makes textual information processing increasingly important

**Comment by NancyLi1013** [comment]
Hi @ignacionr 
Thanks for posting this issue.

@strega-nil 
Could you please help take a look?

**Comment by strega-nil** [comment]
We can't. Package managers cannot, generally, have dependencies until such a point as we can ship binaries. We may reconsider, when vcpkg has a shipping binary that will allow us to bootstrap, but until that point this is unlikely to change.

**Comment by ignacionr** [comment]
> We can't. Package managers cannot, generally, have dependencies until such a point as we can ship binaries. We may reconsider, when vcpkg has a shipping binary that will allow us to bootstrap, but until that point this is unlikely to change.

Oh! Zero dependencies is a lovely thing. I will go on and re-phrase this issue as the need to refactor the file and maybe borrow concepts from libraries such as {fmt}.

**Comment by strega-nil** [comment]
Maybe it's a good plan for the future, but it's definitely not the first thing on our plate. if you would like to, I'd be interested in seeing the work!
